### PR TITLE
Bug/connection passwords being saved without encryption

### PR DESCRIPTION
**Pull request by geovannimp**

### CODE REVIEW

File: src/migration/20200907-add_encrypted-columns.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          isNullable: true
+        }),
+        new TableColumn({
+          name: 'sshPassword_decrypted',
+          type: 'varchar',
+          isNullable: true
+        }),
+      ]
+
+      await runner.startTransaction()
```
Comment (rathboma) [inline src/migration/20200907-add_encrypted-columns.js:26]: this might not work (rollback correctly) because sqlite cannot remove columns? Not sure.

Comment (rathboma) [reply src/migration/20200907-add_encrypted-columns.js:26]: I think it's best to break this out into two migrations:
1. Create new columns
2. run the update queries (with transaction)

**Review by rathboma — CHANGES_REQUESTED** [review]
I'm still actually not sure if it's worth encrypting this stuff or just treating the user environment as trusted, but let's go with my original intent!

Good find here with your typescript changes!

I've added a few comments based on the unfortunate limitations of sqlite.

File: src/migration/20200907-add_encrypted-columns.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+      const entities = await SavedConnection.find();
+      entities.forEach(e => {
+        e.password = connections.find(c => c.id == e.id).password_decrypted
+        e.sshKeyfilePassword = connections.find(c => c.id == e.id).sshKeyfilePassword_decrypted
+        e.sshPassword = connections.find(c => c.id == e.id).sshPassword_decrypted
+      });
+      SavedConnection.save(entities);
+
+      await runner.dropColumns('saved_connection', temp_columns)
```
[the remark this answers is not in the document]
Comment (geovannimp) [reply src/migration/20200907-add_encrypted-columns.js:41]: I can test it again, but I'm almost sure that the columns were dropped.

I tested here again and the columns were dropped. Typeorm must have some tricks to remove the columns.

**Comment by geovannimp** [comment]
@rathboma I looked at de migration again and we don't really need a temp collumn. Take a look at the migration again, it's a lot simpler now

**Comment by rathboma** [comment]
Nice

File: src/migration/20200907-add_encrypted-columns.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      await runner.query('UPDATE saved_connection SET password = NULL, sshKeyfilePassword = NULL, sshPassword = NULL;')
+
+      const entities = await SavedConnection.find();
+      entities.forEach(e => {
+        e.password = original_connections.find(c => c.id == e.id).password
+        e.sshKeyfilePassword = original_connections.find(c => c.id == e.id).sshKeyfilePassword
+        e.sshPassword = original_connections.find(c => c.id == e.id).sshPassword
+      });
+
+      SavedConnection.save(entities);
```
Comment (rathboma) [inline src/migration/20200907-add_encrypted-columns.js:19]: I think you need an `await` here!

Comment (geovannimp) [reply src/migration/20200907-add_encrypted-columns.js]: Yeap, fixed! :D

**Comment by rathboma** [comment]
tested. Works great. Legend

**Pull request merged**